src/analysis/calculus/inverse.lean

```diff
@@ -363,6 +379,25 @@ protected lemma inj_on (hf : approximates_linear_on f (f' : E →L[𝕜] F) s c)
   inj_on f s :=
 inj_on_iff_injective.2 $ hf.injective hc
 
+protected lemma surjective [complete_space E]
+  (hf : approximates_linear_on f (f' : E →L[𝕜] F) univ c) (hc : subsingleton E ∨ c < N⁻¹) :
+  surjective f :=
+begin
+  cases hc with hE hc,
+  { haveI : subsingleton F := (equiv.subsingleton_congr f'.to_linear_equiv.to_equiv).1 hE,
+    exact surjective_to_subsingleton _ },
+  { apply forall_of_forall_mem_closed_ball (λ (y : F), ∃ a, f a = y) (f 0) _,
+    have hc' : (0 : ℝ) < N⁻¹ - c, by { rw sub_pos, exact hc },
+    let p : ℝ → Prop := λ R, closed_ball (f 0) R ⊆ set.range f,
+    have hp : ∀ᶠ (r:ℝ) in at_top, p ((N⁻¹ - c) * r),
+    { have hr : ∀ᶠ (r:ℝ) in at_top, 0 ≤ r := eventually_ge_at_top 0,
+      refine hr.mono (λ r hr, subset.trans _ (image_subset_range f (closed_ball 0 r))),
+      refine hf.surj_on_closed_ball_of_nonlinear_right_inverse f'.to_nonlinear_right_inverse hr _,
+      exact subset_univ _ },
+    refine ((tendsto_id.const_mul_at_top hc').frequently hp.frequently).mono _,
+    exact λ R h y hy, h hy },
```

Comment:
In fact, a further golf, optional as before:
```suggestion
  { apply forall_of_forall_mem_closed_ball (λ (y : F), y ∈ set.range f) (f 0),
    have hc' : (0 : ℝ) < N⁻¹ - c, by { rw sub_pos, exact hc },
    refine (tendsto_id.const_mul_at_top hc').frequently _,
    refine (eventually_ge_at_top (0:ℝ)).frequently.mono _,
    refine λ r hr, subset.trans _ (image_subset_range f (closed_ball 0 r)),
    refine hf.surj_on_closed_ball_of_nonlinear_right_inverse f'.to_nonlinear_right_inverse hr _,
    exact subset_univ _ },
```
(I'm sorry that this version seems to have caused a test to break -- I don't know why!)

Reply:
This was transient, apparently.